The gnu-libiconv test is solely for platforms where the inbuf parameter to
iconv(3) is const char **; we have GNU libiconv but use the POSIX char **.

Note that QIconvCodec should not be used when ICU is enabled.

--- origsrc/qtbase-opensource-src-5.5.1/config.tests/unix/iconv/iconv.pro	2015-10-12 23:35:34.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/config.tests/unix/iconv/iconv.pro	2015-10-18 17:02:47.672312100 -0500
@@ -1,3 +1,3 @@
 SOURCES = iconv.cpp
 CONFIG -= qt dylib
-mac|mingw|openbsd|qnx|haiku:LIBS += -liconv
+mac|mingw|openbsd|qnx|haiku|cygwin-*:LIBS += -liconv
--- origsrc/qtbase-opensource-src-5.5.1/src/corelib/codecs/qiconvcodec.cpp	2015-10-12 23:35:31.000000000 -0500
+++ src/qtbase-opensource-src-5.5.1/src/corelib/codecs/qiconvcodec.cpp	2015-10-18 17:03:43.832524300 -0500
@@ -466,7 +466,7 @@ iconv_t QIconvCodec::createIconv_t(const
         init();
 
     iconv_t cd = (iconv_t) -1;
-#if defined(__GLIBC__) || defined(GNU_LIBICONV) || defined(Q_OS_QNX)
+#if defined(__GLIBC__) || defined(GNU_LIBICONV) || defined(Q_OS_QNX) || defined(Q_OS_CYGWIN)
 #if defined(Q_OS_QNX)
     // on QNX the default locale is UTF-8, and an empty string will cause iconv_open to fail
     static const char empty_codeset[] = "UTF-8";
